libraries/Microsoft.Bot.Builder.Dialogs.Adaptive.Runtime/Adapters/MultiplexingAdapter.cs

```diff
@@ -0,0 +1,170 @@
+        {
+            return GetAppropriateAdapter(continuationActivity.ChannelId).ContinueConversationAsync(claimsIdentity, continuationActivity, callback, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, Activity continuationActivity, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            return GetAppropriateAdapter(continuationActivity.ChannelId).ContinueConversationAsync(claimsIdentity, continuationActivity, audience, callback, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, ConversationReference reference, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            return GetAppropriateAdapter(reference.ChannelId).ContinueConversationAsync(claimsIdentity, reference, callback, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override Task ContinueConversationAsync(ClaimsIdentity claimsIdentity, ConversationReference reference, string audience, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            return GetAppropriateAdapter(reference.ChannelId).ContinueConversationAsync(claimsIdentity, reference, audience, callback, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override Task CreateConversationAsync(string botAppId, string channelId, string serviceUrl, string audience, ConversationParameters conversationParameters, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            return GetAppropriateAdapter(channelId).CreateConversationAsync(botAppId, channelId, serviceUrl, audience, conversationParameters, callback, cancellationToken);
+        }
+
+        /// <inheritdoc/>
+        public override Task<InvokeResponse> ProcessActivityAsync(ClaimsIdentity claimsIdentity, Activity activity, BotCallbackHandler callback, CancellationToken cancellationToken)
+        {
+            return GetAppropriateAdapter(activity.ChannelId).ProcessActivityAsync(claimsIdentity, activity, callback, cancellationToken);
+        }
+
+        /// <summary>
+        /// Maps the incoming ChannelId or route (api/{route}) to the correct adapter. By default, this uses the AdapterSettings specified in appsettings.json.
+        /// Override this method to customize the mapping behavior.
+        /// </summary>
+        /// <param name="adapters">The adapters to map the routes to.</param>
+        protected virtual void BuildAdapterMap(IEnumerable<IBotFrameworkHttpAdapter> adapters)
+        {
+            var adapterSettings = _configuration.GetSection(AdapterSettings.AdapterSettingsKey).Get<List<AdapterSettings>>() ?? new List<AdapterSettings>();
```

Comment:
Note: I believe `AdapterSettings` was the only reason that `MultiplexingAdapter` was placed in the `Adaptive.Runtime` library as opposed to `Integrations`, which would have caused a circular dependency.